io/eolearn/io/sentinelhub_service.py

```diff
@@ -65,23 +66,28 @@ class SentinelHubOGCInput(EOTask):
     :param raise_download_errors: `True` if any errors during download should stop the task and `False` if the task
         should continue and remove timestamps with frames that failed to download.
     :type raise_download_errors: bool
+    :param data_source: A deprecated alternative to data_collection
+    :type data_source: DataCollection
     """
 
-    def __init__(self, layer, feature=None, valid_data_mask_feature='IS_DATA', service_type=None, data_source=None,
+    def __init__(self, layer, feature=None, valid_data_mask_feature='IS_DATA', service_type=None, data_collection=None,
                  size_x=None, size_y=None, maxcc=None, image_format=MimeType.TIFF_d32f, instance_id=None,
-                 custom_url_params=None, time_difference=None, raise_download_errors=True):
+                 custom_url_params=None, time_difference=None, raise_download_errors=True, data_source=None):
+        # pylint: disable=too-many-arguments
```

Comment:
It is a bad indicator that we have to disable pylint warnings like this :)

Reply:
Notice that `SentinelHubInputTask.__init__` is explicitly passing arguments to `SentinelHubInputBase.__init__`. Instead `SentinelHubInputTask` could also use and pass `**kwargs`. This way we would have `6` parameters less.

Usually I'm more inclined towards using `**kwargs` but in this case using explicit parameters also has a few advantages. Therefore I don't have a strong opinnion towards any option.

Reply:
Neither do I. The "proper" way would be passing some sort of (config) object, which is so much cleaner also from the perspective of hinting etc, but that would require quite a chunk of work.